src/Product/SearchProvider.php

```diff
@@ -464,7 +464,7 @@ private function hideUselessFacets(array $facets, $totalProducts)
 
             $facet->setDisplayed(
                 // There are two filters displayed
-                $usefulFiltersCount > 1
+                $usefulFiltersCount > Configuration::get('PS_LAYERED_HIDE_OTHERS_FILTERS') ? 1 : 0
```

Comment:
```suggestion
                $usefulFiltersCount > (int) Configuration::get('PS_LAYERED_HIDE_OTHERS_FILTERS')
```

Reply:
But I'm not sure about this condition. Why not skipping this method?

Reply:
Thanks for the suggestion for code improvement. Passing this method is problematic when you have a filter on a page that does not contain any matching products. Ex: Category when you are in a category that has no children.

Reply:
@PierreRambaud good luck to anyone understanding this after some time :D `$usefulFiltersCount > (int) Configuration::get('PS_LAYERED_HIDE_OTHERS_FILTERS')`

Reply:
I will improve it later for a better understanding ;)